rm/src/main/java/io/seata/rm/RMClient.java

```diff
@@ -34,7 +44,32 @@
     public static void init(String applicationId, String transactionServiceGroup) {
         RmRpcClient rmRpcClient = RmRpcClient.getInstance(applicationId, transactionServiceGroup);
         rmRpcClient.setResourceManager(DefaultResourceManager.get());
-        rmRpcClient.setClientMessageListener(new RmMessageListener(DefaultRMHandler.get(), rmRpcClient));
+        AbstractRMHandler handler = DefaultRMHandler.get();
+
+        Map<Integer, Pair<NettyProcessor, Boolean>> processorMap = new HashMap<>();
+        // rm client handle branch commit processor
+        Pair<NettyProcessor, Boolean> branchCommitProcessor =
```

Comment:
Whether or not to use a user thread pool can be determined based on the request/response type?

Reply:
Phase4 pr has already done so, I now migrate the phase4 related logic to the current pr.